apps/console/src/features/core/components/protected-route.tsx

```diff
@@ -39,20 +37,6 @@ export const ProtectedRoute: FunctionComponent<RouteProps> = (props: RouteProps)
 
     const isAuthenticated: boolean = useSelector((state: AppState) => state.auth.isAuthenticated);
```

Comment:
AFAIR, this logic was added to avoid navigating to 404 pages etc, over and over again. Will removing it affect the functionality?

Reply:
I've revamp the logic of setting the `auth_callback_url` in https://github.com/wso2/identity-apps/pull/3590/commits/14862377d2af41b0ff1c8ab4d4c64b4c25888157. Now it's only persist the auth_callback_url on either first hit of the page or on page refresh. Apart from those two, if the page is either login page, unauthorized page, 404 page or storing_data_disabled page - we will set the `auth_callback_url` to the home page to ensure that those pages won't be coming on the redirection.